authentication-ldap/src/main/java/com/synopsys/integration/alert/authentication/ldap/database/accessor/LDAPConfigAccessor.java

```diff
@@ -0,0 +1,130 @@
+package com.synopsys.integration.alert.authentication.ldap.database.accessor;
+
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.synopsys.integration.alert.api.common.model.exception.AlertConfigurationException;
+import com.synopsys.integration.alert.authentication.ldap.database.configuration.LDAPConfigurationEntity;
+import com.synopsys.integration.alert.authentication.ldap.database.configuration.LDAPConfigurationRepository;
+import com.synopsys.integration.alert.authentication.ldap.model.LDAPGlobalConfigModel;
+import com.synopsys.integration.alert.common.persistence.accessor.UniqueConfigurationAccessor;
+import com.synopsys.integration.alert.common.rest.AlertRestConstants;
+import com.synopsys.integration.alert.common.security.EncryptionUtility;
+import com.synopsys.integration.alert.common.util.DateUtils;
+
+@Component
+public class LDAPConfigAccessor implements UniqueConfigurationAccessor<LDAPGlobalConfigModel> {
+    private final EncryptionUtility encryptionUtility;
+    private final LDAPConfigurationRepository ldapConfigurationRepository;
+
+    public LDAPConfigAccessor(EncryptionUtility encryptionUtility, LDAPConfigurationRepository ldapConfigurationRepository) {
+        this.encryptionUtility = encryptionUtility;
+        this.ldapConfigurationRepository = ldapConfigurationRepository;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public Optional<LDAPGlobalConfigModel> getConfiguration() {
+        return ldapConfigurationRepository.findByName(AlertRestConstants.DEFAULT_CONFIGURATION_NAME).map(this::toModel);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public boolean doesConfigurationExist() {
+        return ldapConfigurationRepository.existsByName(AlertRestConstants.DEFAULT_CONFIGURATION_NAME);
+    }
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRED)
+    public LDAPGlobalConfigModel createConfiguration(LDAPGlobalConfigModel ldapGlobalConfigModel) throws AlertConfigurationException {
+        if (doesConfigurationExist()) {
+            throw new AlertConfigurationException(String.format("A config with the name '%s' already exists.", ldapGlobalConfigModel.getName()));
```

Comment:
Since the name should always be AlertRestConstants.DEFAULT_CONFIGURATION_NAME, I think the message here can be explicit and refer to the LDAP config. Perhaps a message along the lines of "An LDAP configuration already exists."